src/wt3/clients/signal.py

```diff
@@ -1,311 +1,306 @@
-                        
-            except (aiohttp.ClientError, SignalValidationError) as e:
-                logger.warning(f"Fallback attempt {retry_count + 1} failed: {str(e)}")
-                last_error = e
-            except Exception as e:
-                logger.warning(f"Fallback attempt {retry_count + 1} failed with unexpected error: {str(e)}")
-                last_error = e
-
-            retry_count += 1
-            if retry_count < max_retries:
-                wait_time = retry_delay * (2 ** (retry_count - 1))
-                logger.info(f"Retrying fallback in {wait_time:.1f} seconds...")
-                await asyncio.sleep(wait_time)
-        
-        logger.error(f"All attempts to both primary and fallback services failed. Last error: {str(last_error)}")
-        return None
-
-    async def close(self):
-        """Close the client session."""
-        if self._session and not self._session.closed:
-            await self._session.close()
-            self._session = None
-            logger.info("Signal client session closed")
-
-    async def __aenter__(self):
-        """Async context manager entry."""
-        return self
-
-    async def __aexit__(self, exc_type, exc_val, exc_tb):
-        """Async context manager exit."""
-        await self.close()
-        return None
+"""
+Signal client for the WT3 Agent.
+
+This module provides functionality for retrieving trading signals from the signal service,
+including trade decisions and strategy parameters.
+"""
+
+import logging
+import aiohttp
+import os
+import asyncio
+from typing import Dict, Optional, TypedDict, Literal, List
+
+logger = logging.getLogger(__name__)
+
+class StrategyData(TypedDict):
+    """Trading strategy data structure."""
+    position_size_coin: float
+    leverage: float
+    stop_loss: float
+    stop_loss_levels: Optional[List[float]]
+    signal_id: Optional[str]
+
+
+class TradeDecision(TypedDict):
+    """Trade decision data structure."""
+    action: Literal['buy', 'sell', 'close']
+    coin: str
+    strategy: StrategyData
+
+class SignalResponse(TypedDict):
+    """Complete signal response structure."""
+    timestamp: int
+    trade_decision: Optional[TradeDecision]
+
+class SignalError(Exception):
+    """Base exception for signal client errors."""
+    pass
+
+class SignalValidationError(SignalError):
+    """Exception raised when signal data validation fails."""
+    pass
+
+class SignalServiceError(SignalError):
+    """Exception raised when signal service operations fail."""
+    pass
+
+class SignalClient:
+    """Client for retrieving trading signals from the signal service."""
+    
+    def __init__(self):
+        """Initialize the signal client.
+        
+        Raises:
+            SignalError: If client initialization fails
+        """
+        try:
+            self.base_url = os.getenv("SIGNAL_SERVICE_URL")
+            self.fallback_url = "http://127.0.0.1:8001"

```

Comment:
Should this match the .env.example?

```suggestion
            self.fallback_url = "http://signal-service:8001"

```